algorithms/nautilus/fuzzer/other_hierarflow_routines.cpp

```diff
@@ -0,0 +1,108 @@
+ * 
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses/.
+ */
+/**
+ * @file other_hierarflow_routines.cpp
+ * @brief Definition of unclassified HierarFlow routines of Nautilus.
+ * @author Ricerca Security <fuzzuf-dev@ricsec.co.jp>
+ */
+#include <memory>
+#include <variant>
+#include "fuzzuf/algorithms/nautilus/fuzzer/other_hierarflow_routines.hpp"
+#include "fuzzuf/algorithms/nautilus/fuzzer/queue.hpp"
+#include "fuzzuf/algorithms/nautilus/fuzzer/state.hpp"
+
+
+namespace fuzzuf::algorithm::nautilus::fuzzer::routine::other {
+
+/**
+ * @fn
+ * @brief HierarFlow routine for FuzzLoop (fuzz_loop)
+ */
+NullableRef<HierarFlowCallee<void(void)>> FuzzLoop::operator()(void) {
+  CallSuccessors(); // select_input
+
+  state.cycles_done++;
+
+  return GoToDefaultNext();
+}
+
+/**
+ * @fn
+ * @brief HierarFlow routine for SelectInput (select_input)
+ */
+RSelectInput SelectInput::operator()(void) {
+  std::optional<QueueItem> inp;
+
+  if (state.queue.IsEmpty()) {
+    inp = std::nullopt;
+  } else {
+    // TODO: Use lock when multi-threaded
+    inp = state.queue.Pop();
+  }
+
+  // TODO: any good way to prevent inp from being copied?
+  CallSuccessors(inp); // process_chosen_input_or
+  return GoToDefaultNext(); // update_state
+}
+
+/**
+ * @fn
+ * @brief HierarFlow routine for ProcessInput (process_chosen_input_or)
+ * @param (inp) Queue item to process
+ */
+RProcessInput ProcessInput::operator()(std::optional<QueueItem> inp) {
+  if (inp) {
+
+    /* Corpus exists. Mutate input. */
+    CallSuccessors(inp.value()); // initialize_or
+
+    /* Mark as finished */
+    state.queue.Finished(std::move(inp.value()));
+
+    return GoToParent(); // back to select_input
+
+  } else {
+
+    /* Queue is empty. Generate new input. */
+    return GoToDefaultNext(); // generate_input
+
+  }
+}
+
+/**
+ * @fn
+ * @brief HierarFlow routine for GenerateInput (generate_input)
+ * @param (inp) Empty item. Not used.
+ */
+RGenerateInput GenerateInput::operator()(std::optional<QueueItem>) {
+  for (size_t i = 0; i < state.setting->number_of_generate_inputs; i++) {
+    /* Generate random seed and run it */
+    const NTermID& nonterm = state.ctx.NTID("START");
+    size_t len = state.ctx.GetRandomLenForNT(nonterm);
+    Tree tree = state.ctx.GenerateTreeFromNT(nonterm, len);
+
+    /* Run input without duplication */
```

Comment:
I think this is a typo

Reply:
What do you mean?

Reply:
**with** duplication right?

Reply:
RunWithDedup means "Run Without Duplication." That's confusing.

Reply:
Is it intended? Seems the original implementation has both https://github.com/nautilus-fuzz/nautilus/blob/master/fuzzer/src/fuzzer.rs#L120-L141

And even if it's intended, I prefer names like `RunOnWoDedup` then (w/o can be understood, probably).

Reply:
Modified expression in [3698141](https://github.com/fuzzuf/fuzzuf/pull/23/commits/3698141f2d53e34cbe3f200d7aa28bf4cf88403c)

Reply:
great and really-easy-to-understand comment